java/code/src/com/suse/manager/reactor/messaging/RegisterMinionEventMessageAction.java

```diff
@@ -944,7 +944,7 @@ private String getOsRelease(String minionId, ValueMap grains) {
                 "oel".equalsIgnoreCase(grains.getValueAsString("os")) ||
                 "alibaba cloud (aliyun)".equalsIgnoreCase(grains.getValueAsString("os")) ||
                 "almalinux".equalsIgnoreCase(grains.getValueAsString("os")) ||
-                "amazon".equalsIgnoreCase(grains.getValueAsString("os")) ||
+                ("amazon".equalsIgnoreCase(grains.getValueAsString("os")) && "2".equalsIgnoreCase(grains.getValueAsString("osmajorrelease"))) ||
```

Comment:
On 2023 `/etc/redhat-release` does not exist, and I guess it will not exist in newer versions either.